src/L.Realtime.js

```diff
@@ -10,11 +10,12 @@ L.Realtime = L.Layer.extend({
         updateFeature: function(f, oldLayer) {
             if (!oldLayer) { return; }
 
-            if (f.geometry.type === 'Point') {
+            if (oldLayer.setLatLng) {
                 var c = f.geometry.coordinates;
                 oldLayer.setLatLng([c[1], c[0]]);
                 return oldLayer;
             }
+            return oldLayer;
```

Comment:
Yes, this is a consequence of recreating the feature as a new layer. I'd argue that it's better to lose popups etc. than to potentially show the wrong geometry (coordinates), that this change would mean, though.

A better solution is to detect if the layer has a `setLatLngs` method and use that to update the geometry, analogous to how we use `setLatLng` for points today.

We would have to flip the coordinates in this process, and I'm also unsure if there are gotchas regarding nesting of coordinates, since `Polygon` and `MultiPolygon` can have nested coordinate arrays. There might be logic for handling this in `L.GeoJSON` already that can be reused, but it's not something I've looked at.